var/spack/repos/builtin/packages/pythia6/package.py

```diff
@@ -42,6 +42,10 @@ class Pythia6(CMakePackage):
     # the Pythia6 library.
     variant('root', default=False,
             description='Build extra (non OEM) code to allow use by Root.')
+    variant('sft', default=False,
+            description='Apply extra patch provided by CERN EP-SFT group')
+
+    conflicts('+sft', when='~root')
```

Comment:
Why is this patch only allowed in `+root` builds?

The random number generator part looks quite generic anyway.

So the question also is: Why a variant at all?

Reply:
The `conflics` is because the patch changes ROOT-related code, which is only downloaded if `+root`. If you think the changes are ok to be non-optional - I will make them non-optional.

Reply:
Ohh, just now realizing that the patch is partly touching optionally downloaded stuff.

Just an idea: Maybe split the patch in two parts? One for the main (the random number generator) and the other for `+root` only?